### ISSUE TITLE
Setting._onSettingValueChanged called repeatedly for each change

### ISSUE THREAD

**Issue by fieldOfView** [Ultimaker/Uranium#68]
While debugging an issue, I ran into Setting._onSettingValueChanged which, at least on my install, gets called repeatedly each time a setting is changed. Sometimes it is 20 times, sometimes it is 90+ times.

This seems strange, so I thought I would mention it. I have not looked too far into it yet.

**Comment by awhiemstra** [comment]
This is quite possible, the _onSettingValueChanged is a bit of a hack to make updating enabled etc. values work. It gets emitted any time a setting changes value, so changing a parent setting's value can trigger multiple settingValueChanged while the children get updated.

I actually want to pull those properties into a separate object but did not have a good plan for that yet, especially with all the other changes going on with profiles.

**Comment by daid** [comment]
Did some tracing on this. As I noticed it with a very simple settings file with just 3 settings, where 1 setting depended on another, but only on certain conditions.

I noticed when I changed the "main" setting, the setting that has an "inherit_function" using that setting also fires the "settingValueChanged" signal. Even if the actual value never changes.

Traced it down to this signal emit:
https://github.com/Ultimaker/Uranium/blob/master/UM/Settings/Setting.py#L462
Which signals that the default value changed, even if the default was still the same (same goes for the enabledChanged signal)
